go.mod

```diff
@@ -8,19 +8,19 @@ require (
 	github.com/go-git/go-git/v5 v5.1.0
 	github.com/go-logr/logr v1.2.0
 	github.com/google/go-cmp v0.5.6
-	github.com/prometheus/client_golang v1.11.0
-	golang.org/x/crypto v0.0.0-20210817164053-32db794688a5
-	golang.org/x/tools v0.1.6-0.20210820212750-d4cc65f0b2ff
-	k8s.io/api v0.23.0
-	k8s.io/apimachinery v0.23.0
-	k8s.io/cli-runtime v0.23.0
-	k8s.io/client-go v0.23.0
-	k8s.io/klog/v2 v2.30.0
-	k8s.io/kubectl v0.22.2
+	github.com/prometheus/client_golang v1.12.1
+	golang.org/x/crypto v0.0.0-20220214200702-86341886e292
+	golang.org/x/tools v0.1.10-0.20220218145154-897bd77cd717
+	k8s.io/api v0.24.0
+	k8s.io/apimachinery v0.24.0
+	k8s.io/cli-runtime v0.24.0
+	k8s.io/client-go v0.24.0
+	k8s.io/klog/v2 v2.60.1
+	k8s.io/kubectl v0.24.0
```

Comment:
Looks like the bump in the kubectl package version has resulted in some test failures due to changes in the api.

Looks to be that the changes in the api affect these lines:
https://github.com/kubernetes-sigs/kubebuilder-declarative-pattern/blob/e5a105c6e17b2f267a5d79b130334029db898aad/pkg/patterns/declarative/pkg/applier/direct.go#L45

Has changed to the signature:
```go
 v, err := cmdutil.NewFactory(&genericclioptions.ConfigFlags{}).Validator(validationDirective string, verifier *resource.QueryParamVerifier)
```

It seems like the validationDirective options are: https://github.com/kubernetes/apimachinery/blob/97e5df2d0258ac077093867fabf508c1fc31f53b/pkg/apis/meta/v1/types.go#L515-L522

Here is more information on the `resources.QueryParamVerifier`: https://pkg.go.dev/k8s.io/cli-runtime/pkg/resource#QueryParamVerifier

It also seems like the `NewApplyOptions` function was removed, so this line will also need to be modified:
https://github.com/kubernetes-sigs/kubebuilder-declarative-pattern/blob/e5a105c6e17b2f267a5d79b130334029db898aad/pkg/patterns/declarative/pkg/applier/direct.go#L68

The godocs for the `apply` package that the `NewApplyOptions` function came from can be found here: https://pkg.go.dev/k8s.io/kubectl@v0.24.0/pkg/cmd/apply